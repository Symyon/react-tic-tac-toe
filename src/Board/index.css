.Board {
    display: flex;
    flex: 1;
    justify-content: space-between;
    align-content: space-between;
    flex-wrap: wrap;
    background-color: yellowgreen;
    width: 20rem;
    height: 20rem;
    align-self: center;
    border: solid;
    border-color: yellowgreen;
}