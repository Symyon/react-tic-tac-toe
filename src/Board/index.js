import React, { Component } from 'react';
import Cell from '../Cell'
import './index.css';

const cellsArray = [0, 1, 2, 3, 4, 5, 6, 7, 8];

class Board extends Component {
    render() {
        return (
            <div className="Board" style={{ display: 'flex', flex: 1, flexWrap: 'wrap', backgroundColor: 'red', width: 400, height: 400, alignSelf: 'center' }}>
                {
                    cellsArray.map(item => <Cell key={item} order={item} />)
                }
            </div>
        );
    }
}

export default Board;