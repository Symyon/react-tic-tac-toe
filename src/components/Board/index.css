.Board {
    position: relative;
    display: flex;
    flex: 1;
    justify-content: space-between;
    align-content: space-between;
    flex-wrap: wrap;
    background-color: yellowgreen;
    width: 20rem;
    height: 20rem;
    align-self: center;
    border: solid;
    border-color: yellowgreen;
}

.overlay {
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 10;
    background-color: rgba(0,0,0,0.5);
  }

  .overoverlay {
    display: flex;
    flex: 1;
    flex-direction: row;
    height: 20%;
    background-color: rgba(0,0,0,0.75);
    justify-content: center;
    align-items: center;
  }

  .winnerText {
      color: white;
      font-size: 2em;
      margin: 1em;
  }

  .winnerIcon {      
      height: 80%;
      width: auto;
  }