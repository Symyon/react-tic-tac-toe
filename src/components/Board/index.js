import React, { Component } from 'react';
import Cell from '../Cell'
import nought from './img/o.png'
import cross from './img/x.png'
import { EMPTY, CROSS, DRAW } from '../../constants';
import './index.css';

class Board extends Component {
    render() {
        const { game, onClick, winningCells, winner } = this.props;
        return (
            <div>
                <div className="Board">
                    {game && game.map((item, index) => <Cell key={index} index={index} mark={item} onClick={onClick} markIt={winningCells.includes(index)} />)}
                    {winner !== undefined && winner !== EMPTY && <div className="overlay">
                        <div className="overoverlay">
                            <span className="winnerText">{winner === DRAW ? 'Draw' : 'Winner'}</span>
                            {winner !== DRAW && <img src={winner === CROSS ? cross : nought} alt='winner' className='winnerIcon' />};
                        </div>
                    </div>}
                </div>
            </div>
        );
    }
}

export default Board;