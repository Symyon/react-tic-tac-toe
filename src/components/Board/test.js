import React from 'react';
import { shallow } from 'enzyme';
import Board from './';

describe('Board', () => {
    it('render correctly without props', () => {
        const component = shallow(<Board />);
        expect(component).toMatchSnapshot();
    })
});