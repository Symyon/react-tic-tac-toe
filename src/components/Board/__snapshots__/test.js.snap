// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board render correctly without props 1`] = `
<div>
  <div
    className="Board"
  />
</div>
`;

exports[`Board renders vorrectly the overlay when a draw 1`] = `
<div>
  <div
    className="Board"
  >
    <div
      className="overlay"
    >
      <div
        className="overoverlay"
      >
        <span
          className="winnerText"
        >
          Draw
        </span>
        ;
      </div>
    </div>
  </div>
</div>
`;

exports[`Board renders vorrectly the overlay when crosses are winners 1`] = `
<div>
  <div
    className="Board"
  >
    <div
      className="overlay"
    >
      <div
        className="overoverlay"
      >
        <span
          className="winnerText"
        >
          Winner
        </span>
        <img
          alt="winner"
          className="winnerIcon"
          src="x.png"
        />
        ;
      </div>
    </div>
  </div>
</div>
`;

exports[`Board renders vorrectly the overlay when noughts are winners 1`] = `
<div>
  <div
    className="Board"
  >
    <div
      className="overlay"
    >
      <div
        className="overoverlay"
      >
        <span
          className="winnerText"
        >
          Winner
        </span>
        <img
          alt="winner"
          className="winnerIcon"
          src="o.png"
        />
        ;
      </div>
    </div>
  </div>
</div>
`;
